src/sqlparser/src/tokenizer.rs

```diff
@@ -373,6 +376,21 @@ impl<'a> Tokenizer<'a> {
                         }
                     }
                 }
+                x @ 'e' | x @ 'E' => {
+                    chars.next(); // consume, to check the next char
+                    match chars.peek() {
+                        Some('\'') => {
+                            // E'...' - a <character string literal>
+                            let s = self.tokenize_single_quoted_string(chars)?;
```

Comment:
There is a subtle difference between single quoted string and escaped string:
> Also, a single quote can be included in an escape string by writing `\'`, in addition to the normal way of `''`.

In PostgreSQL:
```
test=# select 'a\'z;
 z  
----
 a\
(1 row)

test=# select e'a\'z';                                                                                                                                       ?column? 
----------
 a'z
(1 row)
```